sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs

```diff
@@ -51,7 +51,7 @@ public partial class GetAllFeedbackOptions
         public GetAllFeedbackOptions() { }
         public System.DateTimeOffset? EndTime { get { throw null; } set { } }
         public Azure.AI.MetricsAdvisor.Models.FeedbackType? FeedbackType { get { throw null; } set { } }
-        public Azure.AI.MetricsAdvisor.Models.DimensionKey Filter { get { throw null; } }
+        public Azure.AI.MetricsAdvisor.Models.DimensionKey Filter { get { throw null; } set { } }
```

Comment:
In some types, we were initializing `DimensionKey` in the constructor and letting users add columns manually with `AddDimensionColumn`, similar to what we do to other collections.
```cs
var options = new GetAllFeedbackOptions();
options.Filter.AddDimensionColumn(...);
options.Filter.AddDimensionColumn(...);
```

In other types, we initialized the key to `null` and expected customers to initialize it with a constructor call. There was no consistency at all.
```cs
var options = new OtherOptionsType();
options.DimensionKey = new DimensionKey();
options.DimensionKey.AddDimensionColumn(...);
options.DimensionKey.AddDimensionColumn(...);
```

Given that `DimensionKey` is now immutable after creation, we won't instantiate it for users anymore. We expect them to pass the dimension columns during instantiation:
```cs
var options = new GetAllFeedbackOptions();
options.Filter = new DimensionKey(dimensionColumns);
```